### ISSUE TITLE
conv1d needs additional shape checks

### ISSUE THREAD

**Issue by soumith** [pytorch/pytorch#4607]
```
   import torch.nn as nn
   from torch import autograd
   input = autograd.Variable(torch.randn(1, 1, 2))
   conv = nn.Conv1d(1, 1, 3, stride=2)
   conv(input)
```

Should error out but returns without error (returns uninitialized output).
Reported here: https://discuss.pytorch.org/t/minimum-kernel-size-convolution-solved-by-striding/12162

**Comment by sighingnow** [comment]
I'll do this!

**Comment by soumith** [comment]
fixed via #4637
